adapters/criteo/criteo.go

```diff
@@ -67,9 +67,18 @@ func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.R
 
 	for _, seatBid := range response.SeatBid {
 		for i := range seatBid.Bid {
+			var bidExt openrtb_ext.ExtBid
+			unmarshalExtErr := json.Unmarshal(seatBid.Bid[i].Ext, &bidExt)
+			if unmarshalExtErr != nil {
+				err := &errortypes.BadServerResponse{
+					Message: fmt.Sprintf("Missing ad type in response: %d.", responseData.StatusCode),
```

Comment:
I do not think you should set `responseData.StatusCode` in the error here. The reason is that value will be 200 by the time code reaches here.